activemq-client/src/main/java/org/apache/activemq/ActiveMQProducer.java

```diff
@@ -0,0 +1,546 @@
+
+    @Override
+    public double getDoubleProperty(String name) {
+        Object value = getCreatedMessageProperties().get(name);
+        if (value == null) {
+            throw new NullPointerException("property " + name + " was null");
+        }
+        Double rc = (Double)TypeConversionSupport.convert(value, Double.class);
+        if (rc == null) {
+            throw new MessageFormatRuntimeException("Property " + name + " was a " + value.getClass().getName() + " and cannot be read as a double");
+        }
+        return rc.doubleValue();
+    }
+
+    @Override
+    public String getStringProperty(String name) {
+        Object value = getCreatedMessageProperties().get(name);
+        if (value == null) {
+            throw new NullPointerException("property " + name + " was null");
+        }
+        String rc = (String)TypeConversionSupport.convert(value, String.class);
+        if (rc == null) {
+            throw new MessageFormatRuntimeException("Property " + name + " was a " + value.getClass().getName() + " and cannot be read as a string");
+        }
+        return rc;
+    }
+
+    @Override
+    public Object getObjectProperty(String name) {
+        if (name == null) {
+            throw new NullPointerException("Property name cannot be null");
+        }
+
+        // TODO: Update PropertyExpression to handle converting message headers to properties for JMSProducer.
+        //PropertyExpression expression = new PropertyExpression(name);
+        //return expression.evaluate(this);
+        return getCreatedMessageProperties().get(name);
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return getCreatedMessageProperties().keySet();
+    }
+
+    @Override
+    public JMSProducer setJMSCorrelationIDAsBytes(byte[] correlationID) {
+        if(correlationID != null) {
+            this.correlationIdBytes = Arrays.copyOf(correlationID, correlationID.length);
+        }
+        return this;
+    }
+
+    @Override
+    public byte[] getJMSCorrelationIDAsBytes() {
+        return this.correlationIdBytes;
+    }
+
+    @Override
+    public JMSProducer setJMSCorrelationID(String correlationID) {
+        this.correlationId = correlationID;
+        return this;
+    }
+
+    @Override
+    public String getJMSCorrelationID() {
+        return this.correlationId;
+    }
+
+    @Override
+    public JMSProducer setJMSType(String type) {
+        this.type = type;
+        return this;
+    }
+
+    @Override
+    public String getJMSType() {
+        return this.type;
+    }
+
+    @Override
+    public JMSProducer setJMSReplyTo(Destination replyTo) {
+        this.replyTo = replyTo;
+        return this;
+    }
+
+    @Override
+    public Destination getJMSReplyTo() {
+        return replyTo;
+    }
+
+    private void checkPropertyName(String name) {
+        if(name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("Property name cannot be empty or null");
+        }
```

Comment:
I was not able to find any references where ActiveMQ performs these checks today. I'm starting to implement them. If there is a util that already does it for the JMS v1.1 side, please let me know and I'll re-use that to avoid dupe.

A 'find' across source tree only matched LIKE and BETWEEN in the selector parser (which isn't used by the v1.1 producer)

Reply:
I don't think there are any checks in the 1.1 client which isn't surprising given it tends to play a bit fast and loose with the spec requirements.  There are checks already in Qpid JMS you could look at and leverage, along with configuration at the connection level to disable them for those that simply can't live within the confines of the spec requirements.  The older client should probably be made more spec compliant but that's a whole other can of worms.

Reply:
Got it. I placed the property verification checks in the ActiveMQMessageProducerSupport class with that in mind-- we can retrofit JMS 1.1 down the line. These changes have been pushed into the latest rev of PR (b97ed9d), and I have added tasks as reminders to add unit test coverage.